external_parser/event_processors/joined_event.h

```diff
@@ -374,19 +366,10 @@ struct ca_joined_event : public typed_joined_event {
 
     example *ex = examples[0];
     ex->l.cb_cont.costs.push_back(
-        {interaction_data.action, 0.f, interaction_data.pdf_value});
+        {interaction_data.action, -1.f * reward, interaction_data.pdf_value});
     return true;
   }
 
-  void set_cost(v_array<example *> &examples, float reward,
-          size_t index = 0) const override {
-    if (std::isnan(interaction_data.action)) {
-      return;
-    }
-
-    examples[index]->l.cb_cont.costs[0].cost = -1.f * reward;
-  }
-
   bool should_calculate_reward(
```

Comment:
so all the `should_calculate_reward`'s functionality has been moved to the reward functions, except for cats?

Reply:
forgot to remove this one, `should_calculate_reward` functionality should be handled by reward functions for all problem types.